libs/src/services/discoveryProvider/DiscoveryProvider.ts

```diff
@@ -154,13 +169,34 @@ export class DiscoveryProvider {
     this.maxRequestsForTrue404 = MAX_MAKE_REQUEST_RETRIES_WITH_404
 
     this.monitoringCallbacks = monitoringCallbacks
+    this.discoveryNodeSelector = discoveryNodeSelector
+    this.discoveryNodeMiddleware = discoveryNodeSelector?.createMiddleware()
   }
 
   async init() {
-    const endpoint = await this.serviceSelector.select()
-    this.setEndpoint(endpoint)
+    if (this.discoveryNodeSelector) {
+      this.discoveryNodeSelector.addEventListener(
+        'change',
+        (endpoint: string) => {
+          this.setEndpoint(endpoint)
+        }
+      )
+
+      const endpoint = await this.discoveryNodeSelector.getSelectedEndpoint()
+      if (endpoint) {
```

Comment:
nit: Not sure we need the `if` guard here

Reply:
so apparently it returns string | undefined

Reply:
Does it need to be `string | null`?

Reply:
yeah for the setter, but can check

Reply:
weird - `getSelectedEndpoint` _should_ return `string | null` 🤔

Reply:
Oh oh I see, the setter expects explicitly a string, no nulls no undefineds

Reply:
weird.. so the old select must not be typed correctly because it too can return `null`

Reply:
Will update that one

Reply:
Probably not necessary, just an observation!